src/containers/Tenant/Diagnostics/TopQueries/TopQueries.tsx

```diff
@@ -56,23 +69,60 @@ export const TopQueries = ({path, type, changeSchemaTab}: TopQueriesProps) => {
         wasLoaded,
         error,
         data: {result: data = undefined} = {},
+        filters: storeFilters,
     } = useTypedSelector((state) => state.executeTopQueries);
 
+    const preventUpdate = useRef(false);
+
+    const [filters, setFilters] = useState<ITopQueriesFilters>(storeFilters);
```

Comment:
Is it necessary to use internal component state for filters? May be it is possible to use redux state as one place for storing actual filters?

Reply:
Component state handles default values. Some of the filters sync between url params and the redux state. Url should preserve modified state, but should not capture the default state.